src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2219,10 +2224,22 @@ because xbuild doesn't support framework reference assemblies.
   </CreateItem>
 </Target>
 
+<Target Name="_PrepareNativeAssemblySources">
+  <ItemGroup>
+    <_BuiltAbis Include="$(_BuildTargetAbis)" />
+    <_TypeMapAssemblySource Include="$(_NativeAssemblySourceDir)typemap.jm.%(_BuiltAbis.Identity).s">
+      <abi>%(_BuiltAbis.Identity)</abi>
+    </_TypeMapAssemblySource>
+    <_TypeMapAssemblySource Include="$(_NativeAssemblySourceDir)typemap.mj.%(_BuiltAbis.Identity).s">
+      <abi>%(_BuiltAbis.Identity)</abi>
+    </_TypeMapAssemblySource>
+  </ItemGroup>
+</Target>
+
 <Target Name="_GenerateJavaStubs"
-  DependsOnTargets="_SetLatestTargetFrameworkVersion;_PrepareAssemblies;$(_AfterPrepareAssemblies)"
+  DependsOnTargets="_SetLatestTargetFrameworkVersion;_PrepareAssemblies;$(_AfterPrepareAssemblies);_PrepareNativeAssemblySources"
   Inputs="$(MSBuildAllProjects);@(_ResolvedAssemblies);$(_AndroidManifestAbs);$(_AndroidBuildPropertiesCache);@(_AndroidResourceDest)"
-  Outputs="$(_AndroidStampDirectory)_GenerateJavaStubs.stamp">
+  Outputs="@(_TypeMapAssemblySource);$(_AndroidStampDirectory)_GenerateJavaStubs.stamp">
```

Comment:
I don't think we need to change the `Outputs` here, the stamp file should be sufficient.